### PR TITLE
Improve MonaiAlgo

### LINKED ISSUE #4982: Improve MonaiAlgo: weight loading warning, disable ckpt loaders, support sending best model, determinism
**Is your feature request related to a problem? Please describe.**
A couple of things to improve MonaiAlgo

**Describe the solution you'd like**

1. warn if no weights were loaded
2. disable checkpoint loaders by default if defined in train/evaluate configs
3. send best checkpoint if requested
4. allow deterministic training

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by holgerroth**
Fixes #4982.

### Description
Several improvements to MonaiAlgo:

1. warn if no weights were loaded
2. disable checkpoint loaders by default if defined in train/evaluate configs
3. send best checkpoint if requested
4. allow deterministic training

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/fl/client/monai_algo.py [hunk 1]
```diff
@@ -245,26 +267,46 @@ def get_weights(self, extra=None):
...
+            or "best" local weights if `extra` has `ExtraItems.MODEL_NAME` with string containing "best".
 
         """
         if extra is None:
             extra = {}
+
+        # by default return current weights, return best if requested via model name.
+        return_best = False
+        if ExtraItems.MODEL_NAME in extra:
+            if "best" in extra.get(ExtraItems.MODEL_NAME).lower():
```
Comment (Nic-Ma) [inline monai/fl/client/monai_algo.py:280]: I feel it's not a robust way to determine the behavior by some string comparison.
Better to have a more clear argument in the `ExtraItems` in the `extra`, for example: `ExtraItems.MODEL_TYPE`.
@wyli What do you think?

Thanks.

File: monai/fl/client/monai_algo.py [hunk 2]
```diff
@@ -178,9 +197,10 @@ def initialize(self, extra=None):
         if BundleKeys.TRAIN_TRAINER_MAX_EPOCHS in self.train_parser:
             self.train_parser[BundleKeys.TRAIN_TRAINER_MAX_EPOCHS] = self.local_epochs
 
-        self.train_parser.parse()
-        self.eval_parser.parse()
-        self.filter_parser.parse()
+        # remove checkpoint loaders
+        if self.disable_ckpt_loader:
+            remove_ckpt_loader(self.train_parser)
+            remove_ckpt_loader(self.eval_parser)
```
[the remark this answers is not in the document]
Comment (holgerroth) [reply monai/fl/client/monai_algo.py:203]: That's a very good point. Let me test this approach.

[the remark this answers is not in the document]
Comment (holgerroth) [reply monai/fl/client/monai_algo.py:214]: changed and tested.

File: monai/fl/client/monai_algo.py (hunk 1 above)
Comment (holgerroth) [reply monai/fl/client/monai_algo.py]: Changed to using `ModelType` for best and final model.

**Review by wyli — APPROVED** [review]
Thanks, it looks good to me.

**Review by Nic-Ma — COMMENTED** [review]
Looks good to me.

Thanks for the update.

**Pull request merged**